expression/builtin.go

```diff
@@ -657,8 +657,8 @@ type builtinFunc interface {
 	// getArgs returns the arguments expressions.
 	getArgs() []Expression
 	// isDeterministic checks if a function is deterministic.
-	// A function is deterministic if it returns same results for same inputs.
-	// e.g. random is non-deterministic.
+	// A function is deterministic if it returns the same result whenever it's called with the same inputs.
```

Comment:
The comment needs changes. We set `isDeterministic=true` for `NOW()` is not because of "it returns the same result whenever it's called with the same inputs.".

Reply:
already discussed with @XuHuaiyu before.

`isDeterministic()` is only used in constant folding, my point is we can change `isDeterministic()` to `canBeFolded()` and change `baseBuiltinFunc.deterministic` to `baseBuiltinFunc.canBeFolded`

Reply:
@zz-jason Agree with this naming change (defer to another PR).